src/ambianic/pipeline/ai/face_detect.py

```diff
@@ -73,3 +80,27 @@ def process_sample(self, **sample):
                               'Dropping sample: %r',
                               e,
                               sample)
+
+    def convert_inference_result(self, inference_result):
+
+        inf_json = []
+        if inference_result:
+            for inf in inference_result:
+                label, confidence, box = inf[0:3]
+                log.info('label: %s , confidence: %.0f, box: %s',
+                         label,
+                         confidence,
+                         box)
+                one_inf = {
+                    'label': label,
+                    'confidence': float(confidence),
```

Comment:
@bhavikapanara instead of looking for all the possible places that a numpy type serialization can occur and hardcoding a type conversion, which is error prone and not too elegant, I suggest looking at more robust solutions. For example finding a library that can handle json serialization of numpy types such as [this one](https://github.com/mverleg/pyjson_tricks). There may be other alternatives. Please research.

Reply:
Looks like [encoding numpy scalars is an issue](https://json-tricks.readthedocs.io/en/latest/#numpy-scalars) for json-tricks.

The numpy team is also [aware of the issue](https://github.com/numpy/numpy/issues/16432) and there is no agreed upon solution yet.

Looks like adding a simple custom encoder extension is the preferred solution at the moment as presented [1)here](https://stackoverflow.com/a/27050186/12015435) and  [2)here](https://gist.github.com/massgh/297a73f2dba017ffd28dbc34b9a40e90).

Reply:
Ok...Will implement this

Reply:
Here I have added the encoder to parse the numpy scalar.
https://github.com/ambianic/ambianic-edge/blob/4b0c350ab99df9089ae9a11a76fb6735d489267b/src/ambianic/pipeline/store.py#L108

For the test case, 
I have observed that the `_save_sample` function is covered by the try block, so whenever `json.dump()` fails, exception raise. So Assertion won't work here in test cases

https://github.com/ambianic/ambianic-edge/blob/4b0c350ab99df9089ae9a11a76fb6735d489267b/src/ambianic/pipeline/store.py#L157

Reply:
Encoder looks good.

For testing this type of cases, see the PipelineElement [mockup with callback registration](https://github.com/ambianic/ambianic-edge/blob/4b0c350ab99df9089ae9a11a76fb6735d489267b/tests/pipeline/ai/test_face_detect.py#L63) technique we use in the same test file. Think how you can implement a similar callback around `_save_sample` to make sure that the new encoder works in context.  

I would also suggest adding a few unit tests just around the encoder. A code that failed before with the default encoder (assert failure) should not fail with the new encoder (assert expected result).

Reply:
Ok.
Working on this

Reply:
OK, I am merging this PR. Please submit encoder unit tests as a new PR.